src/edu/ucsb/cs56/projects/games/fish_animation/FishAnimationEnvironment.java

```diff
@@ -513,7 +528,11 @@ public void display(int delay, DrawingPanel fishPanel)
 					// jellyfish and penalize if true
 					if ((xj - 20 > posX - 200 && xj + 20 < posX + 40) && (yj + 60 > posY - 60 && yj < posY + 25)) {
 						eaten -= 5;
-						health -= 10;
+						if (character_type){
```

Comment:
Does it matter that the whale has twice as much health if the shark takes half as much damage?

Reply:
It is because the whlae has more MAXhealth so I hope that it will loss more health as well. Or it may be harder for whale to be died. Besides, whale has lower healing speed. Do I need to make whale loss more than 10 health? Or do you mean that whale and shark should have same health?

Reply:
This is a design question.  Right now, the shark has 50 HP, and the whale has 100 HP.  But the shark takes 5 damage per jellyfish and the whale takes 10 damage per jellyfish.  So:
 - 10 jellyfish kill a shark.
 - 10 jellyfish kill a whale.

Then look at healing.  The shark has 50 HP and heals 2 HP per fish.  The whale has 100 HP and heals 1 HP per fish.  So:
 - A shark needs 25 fish to fully heal.
 - A whale needs 100 fish to fully heal.

When you combine those two measurements, you get:
 - A shark and a whale are equally "tough" when taking damage.
 - A shark heals four times faster than a whale!
 - Therefore, a shark is strictly better than a whale.

So with the code the way you have it now, you should always play as a shark.  The shark is better at everything.

Reply:
Get it. Therefore, I need to balance the two characters, which can make the game fair.

Reply:
What I though right now is to make 5 jellyfish kill a shark. Because shark has faster healing speed. Whale has more health but slower healing speed. It is one kind of balance. How do you feel about it.

Reply:
Nope!  `character_type == true` means it's a shark.  Look at line 532 vs line 534.  That condition says that a shark hitting a jellyfish should lose 5 HP, but a whale hitting a jellyfish should lose 10 HP.  Maybe they should both lose 10 HP?

Reply:
I have not pushed the lastest version. What I want to do is make both of them lose 10Hp

Reply:
It is same as what you said

Reply:
Better... but does this really need an `if` / `else`?

Reply:
I think it is straight. Or do you have any suggestions?

Reply:
Think about it for a minute.  What should this be instead?

```java
if (character_type){
  health -= 10;
}else{
  health -=10;
}
```

Reply:
lol，just health -= 10.my bad, i did not see it carefully.

Reply:
fixed